gunicorn/arbiter.py

```diff
@@ -499,11 +467,11 @@ def murder_workers(self):
             except (OSError, ValueError):
                 continue
 
-            if not worker.aborted:
+            if worker.killed['by'] is None:
                 self.log.critical("WORKER TIMEOUT (pid:%s)", pid)
-                worker.aborted = True
                 self.kill_worker(pid, signal.SIGABRT)
-            else:
+            elif (worker.killed['by'] == signal.SIGABRT
+                  and time.monotonic() > worker.killed['when'] + worker.timeout):
```

Comment:
Yes, what I sought to fix was that the "_You have 2 seconds!_" threat would be honored :) Because right now, it just depends on how often we enter the `manage_workers()` method, which could be more often than once per second.